### ISSUE TITLE
[ETHEREUM-CONTRACTS] [GDA] Audit-related Fixes

### ISSUE THREAD

**Issue by vmichalik** [superfluid-finance/protocol-monorepo#1746]
## Overview
**What?**
Fixes applied to the GDA-related contracts given the audit by Trail of Bits.

**How?**
A review of the Trail of Bits audit and then application of the fixes if deemed reasonable.
Firstly, an internal review of the fixes applied and internal review of the code again.
Then another three engineer days review of the fixes by Trail of Bits before the final review.

**Why?**
To ensure that any issues found by ToB are resolved to mitigate smart contract security risks.

**Deadline:** 04/12/2023

**UI/UX Design:** N/a

## Acceptance criteria
 - [x] Fix suggestions by Trail of Bits are 
 - [x] Satisfies entire acceptance criteria
 - [ ] Consider communication & visibility
 - [ ] Code review is approved by at least 2 reviewers
 - [ ] Tested by Elvijs/QA & discovered bugs resolved
 - [ ] Documentation updated
 - [ ] Assignee/opened by merges PR
 - [ ] Automated E2E tests for happy path
 - [ ] Code is merged
 - [ ] Observability of critical user path
 - [ ] Steps to access code change as a user

## Links/Resources
n/a

## Readiness criteria
- [x] Descriptive title
- [x] What?
- [x] Why?
- [x] How?
- [x] Acceptance criteria complying with Definition of Done
- [ ] Estimate/Story points
- [x] Deadline: If not a part of an epic
- [x] Priority: Reflected as order on the bord, walking the board from the top and right
- [x] Assignee
- [x] Labels
- [x] Link to epic: If belonging to an epic
- [x] A task is always a separate ticket/issue
- [x] UI/UX Design (optional)
- [x] Blockers/Prerequisites (optional): Use Zenhub blocked by/blocking feature

**Comment by philipandersson** [comment]
closes #1729
